proc/proc_linux.go

```diff
@@ -467,9 +467,10 @@ func (dbp *Process) resume() error {
 	}
 	// everything is resumed
 	for _, thread := range dbp.Threads {
-		if err := thread.resume(); err != nil && err != sys.ESRCH {
+		if err := thread.resume(sig); err != nil && err != sys.ESRCH {
```

Comment:
On Tue, Aug 16, 2016 at 8:01 AM Alessandro Arzilli <notifications@github.com>
wrote:

> In proc/proc_linux.go
> <https://github.com/derekparker/delve/pull/613#discussion_r74952533>:
>
> > @@ -467,9 +467,10 @@ func (dbp *Process) resume() error {
> >  	}
> >  	// everything is resumed
> >  	for _, thread := range dbp.Threads {
> > -		if err := thread.resume(); err != nil && err != sys.ESRCH {
> > +		if err := thread.resume(sig); err != nil && err != sys.ESRCH {
>
> Doesn't this deliver one signal per thread instead of just one signal?
>

Yes, but from what I've seen, only one of them appears to be signaled. I
haven't yet figured out if it's possible to select one and only one thread.
My attempts so far haven't worked.

(dlv) b signals.go:15
Breakpoint 1 set at 0x40128e for main.main()
/home/tgrennan/local/signals.go:15
(dlv) c
22831 waiting for SIGCONT(18)...
> main.main() /home/tgrennan/local/signals.go:15 (hits goroutine(1):1
total:1) (PC: 0x40128e)
    10: func main() {
    11:         ch := make(chan os.Signal)
    12:         signal.Notify(ch, syscall.SIGCONT)
    13:         fmt.Printf("%d waiting for SIGCONT(%d)...\n",
    14:                 os.Getpid(), syscall.SIGCONT)
=>  15:         _ = <-ch
    16:         signal.Stop(ch)
    17:         fmt.Println("Done")
    18: }
(dlv) goroutines
[6 goroutines]
* Goroutine 1 - User: /home/tgrennan/local/signals.go:15 main.main
(0x40128e) (thread 22837)
  Goroutine 2 - User: /usr/local/go/src/runtime/proc.go:260 runtime.gopark
(0x4289ca)
  Goroutine 3 - User: /usr/local/go/src/runtime/proc.go:260 runtime.gopark
(0x4289ca)
  Goroutine 4 - User: /usr/local/go/src/runtime/proc.go:260 runtime.gopark
(0x4289ca)
  Goroutine 5 - User: /usr/local/go/src/runtime/sigqueue.go:116
os/signal.signal_recv (0x43b11e)
  Goroutine 6 - User: /usr/local/go/src/runtime/proc.go:260 runtime.gopark
(0x4289ca)
(dlv) threads
  Thread 22831 at 0x4561d3 /usr/local/go/src/runtime/sys_linux_amd64.s:388
runtime.futex
  Thread 22835 at 0x455db7 /usr/local/go/src/runtime/sys_linux_amd64.s:95
runtime.usleep
  Thread 22836 at 0x456204 /usr/local/go/src/runtime/sys_linux_amd64.s:408
runtime.clone
* Thread 22837 at 0x40128e /home/tgrennan/local/signals.go:15 main.main
  Thread 22838 at 0x4561d3 /usr/local/go/src/runtime/sys_linux_amd64.s:388
runtime.futex
(dlv) signal 18
Done
Process 22831 has exited with status 0